src/bin/server.rs

```diff
@@ -56,7 +59,32 @@ fn main() {
 
     let server = if dotenv::var("USE_HYPER").is_ok() {
         println!("Booting with a hyper based server");
-        Hyper(HyperService::new(app, threads as usize))
+        let addr = ([127, 0, 0, 1], port).into();
+        let service = HyperService::new(app, threads as usize);
+        let server = hyper::Server::bind(&addr).serve(service);
+
+        let (tx, rx) = futures::sync::oneshot::channel::<()>();
+        let server = server
+            .with_graceful_shutdown(rx)
+            .map_err(|e| log::error!("Server error: {}", e));
+
+        ctrlc_handler(move || tx.send(()).unwrap_or(()));
+
+        let mut rt = tokio::runtime::Builder::new()
+            .core_threads(4)
+            .name_prefix("hyper-server-worker-")
+            .after_start(|| println!("Stared thread {}", thread::current().name().unwrap_or("?")))
+            .before_stop(|| {
+                println!(
```

Comment:
Would it be better to use the log macro? Some of the pre-existing printlns don't and some do. I am not sure what the rational is.